packages/devtools/lib/src/memory/memory_plotly.dart

```diff
@@ -254,4 +379,140 @@ class MemoryPlotly {
   void setLiveUpdate({bool live}) {
     liveUpdate = live;
   }
+
+  void plotSnapshot() {
+    if (!hasEventTimeline) createEventTimeline();
+
+    final List data = getProperty(_memoryChart.element, 'data');
+    final Data capacityTrace = data[MEMORY_CAPACITY_TRACE];
+    final int timestamp = capacityTrace.x[capacityTrace.x.length - 1];
+
+    eventTimeline.plotSnapshot(timestamp);
+  }
+
+  void plotReset() {
+    if (!hasEventTimeline) createEventTimeline();
+
+    final List data = getProperty(_memoryChart.element, 'data');
+    final Data capacityTrace = data[MEMORY_CAPACITY_TRACE];
+    final int timestamp = capacityTrace.x[capacityTrace.x.length - 1];
+
+    eventTimeline.plotReset(timestamp);
+  }
+}
+
+class EventTimeline {
```

Comment:
add a comment or make the name more clearly memory related?

Reply:
Added comment:

/// Create an Event Timeline subplot, notice it is associated with y2.  This
/// requires that the layout these traces exist in theyaxis2 area.  For an
/// example, look at MemoryPloty's getMemoryLayout method it creates a yaxis2
/// positioned above yaxis.